tests/Integration/Behaviour/Features/Scenario/Carrier/carrier_management.feature

```diff
@@ -175,26 +225,236 @@ Feature: Carrier management
+      | isFree           | true                               |
+      | shippingMethod   | weight                             |
+      | taxRuleGroup     | US-AL Rate (4%)                    |
+      | rangeBehavior    | disabled                           |
+    When I edit carrier "carrier1" called "newCarrier1" with specified properties:
+      | shippingHandling  | true                              |
+    Then carrier "newCarrier1" should have the following properties:
+      | shippingHandling | true                              |
+      | isFree           | false                             |
+
+  Scenario: Partially editing carrier with is free and additional fees already true
+    When I create carrier "carrier1" with specified properties:
+      | name             | Carrier 1                          |
+      | grade            | 1                                  |
+      | trackingUrl      | http://example.com/track.php?num=@ |
+      | position         | 2                                  |
+      | active           | true                               |
+      | delay[en-US]     | Shipping delay                     |
+      | shippingHandling | true                               |
+      | isFree           | false                              |
+      | shippingMethod   | weight                             |
+      | taxRuleGroup     | US-AL Rate (4%)                    |
+      | rangeBehavior    | disabled                           |
+    When I edit carrier "carrier1" called "newCarrier1" with specified properties:
+      | isFree  | true                              |
+    Then carrier "newCarrier1" should have the following properties:
+      | shippingHandling | false                            |
+      | isFree           | true                             |
+
+  Scenario: Partially editing carrier with is free and additional fees at true
+    When I create carrier "carrier1" with specified properties:
+      | name             | Carrier 1                          |
+      | grade            | 1                                  |
+      | trackingUrl      | http://example.com/track.php?num=@ |
+      | position         | 2                                  |
+      | active           | true                               |
+      | delay[en-US]     | Shipping delay                     |
+      | shippingHandling | true                               |
+      | isFree           | false                              |
+      | shippingMethod   | weight                             |
+      | taxRuleGroup     | US-AL Rate (4%)                    |
+      | rangeBehavior    | disabled                           |
+    When I edit carrier "carrier1" called "newCarrier1" with specified properties:
+      | isFree            | true                              |
+      | shippingHandling  | true                              |
+    Then carrier edit should throw an error with error code "INVALID_SHIPPING_HANDLING"
```

Comment:
As suggested in another comment, it's worth having a dedicated error code for this domain constraint